src/orbit.jl

```diff
@@ -141,28 +222,29 @@ function integrate(M::AxisymmetricEquilibrium, gcp::GCParticle, phi0,
 
     # Initial Conditions
     mc2 = gcp.m*c0^2
-    p0 = sqrt(((1e3*e0*gcp.energy + mc2)^2 - mc2^2)/(c0*c0))
+    p0 = sqrt(((1e3*e0*gcp.energy + mc2)^2 - mc2^2)/(c0*c0)) # The initial particle momentum
     # move abs(pitch) away from one to avoid issues when calculating jacobian;
     # don't try to be clever by using clamp it doesn't work with the autodiff
     if abs(gcp.pitch) == 1.0
         pitch0 = sign(gcp.pitch)*prevfloat(abs(gcp.pitch))
     else
         pitch0 = gcp.pitch
     end
-    p_para0 = p0*pitch0*M.sigma
-    p_perp0 = p0*sqrt(1.0 - pitch0^2)
-    mu_0 = (p_perp0^2)/(2*gcp.m*M.b(gcp.r,gcp.z))
+    p_para0 = p0*pitch0*M.sigma # The initial parallel momentum
+    p_perp0 = p0*sqrt(1.0 - pitch0^2) # The initial perpendicular momentum
+    mu_0 = (p_perp0^2)/(2*gcp.m*M.b(gcp.r,gcp.z)) # The initial (and constant) magnetic moment
 
-    r0 = SVector{5}(gcp.r, one(gcp.r)*phi0, gcp.z, one(gcp.r)*p_para0, one(gcp.r)*mu_0)
-    stat.ri = r0
+    r0 = SVector{5}(gcp.r, one(gcp.r)*phi0, gcp.z, one(gcp.r)*p_para0, one(gcp.r)*mu_0) # The initial guiding-centre element vector
+    stat.ri = r0 # The initial status vector is the initial vector
 
-    gc_ode = make_gc_ode(M,gcp,stat,vacuum,drift)
-    stat.vi = gc_ode(r0,false,0.0)
-    stat.initial_dir = abs(stat.vi[1]) > abs(stat.vi[3]) ? 1 : 3
+    gc_ode = make_gc_ode(M,gcp,stat,vacuum,drift) # Make the ordinary differential equation
+    stat.vi = gc_ode(r0,false,tmin) # Obtain the initial velocity
+    stat.initial_dir = abs(stat.vi[1]) > abs(stat.vi[3]) ? 1 : 3 # Is it mostly in the R- or z-direction? Element number 1 or 3?
 
-    tspan = (one(gcp.r)*tmin*1e-6,one(gcp.r)*tmax*1e-6)
```

Comment:
Any reason you changed the units?

Reply:
I thought it was confusing when _dt=cyclotron_period(M,gcp)*1e5_ and _tmin=0.0,tmax=1e6*dt_ were set as the default values in the old version, just to be undone again as _tspan = (one(gcp.r)*tmin*1e-6,one(gcp.r)*tmax*1e-6)_ and _dts = dt*1e-6_ on line 163 and 188 (old version). To me, it makes more sense that, when the default keyword argument values are set, dt is clearly set as a 10th of a cyclotron period (_dt=cyclotron_period(M,gcp)*1e-1_) and tmax explicitly undo that and is set to _1e5*dt_ (i.e. _tmin=0.0,tmax=1e5\*dt*1e1_) as in the suggested new version.